### PR TITLE
Run dotnet-trace in regression test

### PR DESCRIPTION
**Pull request by yufeih**
Starting from .NET 5, we could use `dotnet-trace` to [collect a trace from app start](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-trace#use-diagnostic-port-to-collect-a-trace-from-app-startup), this allows us to profile regression tests on CI server and upload performance trace to azure pipeline artifacts for each code change.

[AB#232284](https://dev.azure.com/ceapex/Engineering/_workitems/edit/232284/)
[AB#372082](https://dev.azure.com/ceapex/Engineering/_workitems/edit/372082)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6979)

### CODE REVIEW

**Comment by yufeih** [comment]
`dotnet-trace` collect from startup causes child process hang: https://github.com/dotnet/diagnostics/issues/1900

File: test/docfx.RegressionTest/SpeedScope.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                .OrderByDescending(item => item.Second)
+                .Select(item => (item.First.name, (int)Math.Round(100 * item.Second / totalTime)))
+                .Take(top)
+                .ToArray();
+
+            static bool IsMyCode(string method)
+            {
+                return method.Contains("!") &&
+                      !method.Contains("?!?") &&
+                      !method.StartsWith("System.", StringComparison.OrdinalIgnoreCase);
```
Comment (mingwli) [inline test/docfx.RegressionTest/SpeedScope.cs:85]: >!method.StartsWith("System.", [](start = 22, length = 29)

put as first condition may increase a little bit perf? (more than one third starts with System. in AspNetCore)

**Review by mingwli — APPROVED** [review]
:shipit:

**Pull request merged**